### PR TITLE
Improve M5Core2TestApp

### PR DESCRIPTION
**Pull request by networkfusion**
## Description
* Sort usings.
* Move const Strings to top.
* Add Vibrate.
* Add Gyro.
* Add Accelerometer.

Before merging the following issues should be discussed:
* Add temperature (noting font issue): In that you cannot load a font from an upstream lib, and testing currently suggests that even using something like `Console.Font = Resources.GetFont(Resources.FontResources.comicsansms16);` from a local project will still not work!

* Improve messages (still considering memory vs readability?!): This lib uses const strings at the expense of readability. We should come up with a better solution and, (using the new benchmark utility) improve the corlib so it does not matter!

## Motivation and Context
General improvements and suggestions.
This PR could be split out if required, but probably requires issues created and notes removed...

## How Has This Been Tested?
Ran against a local device.

## Screenshots
image

## Types of changes

- [x] Improvement (non-breaking change that improves a feature, code or algorithm)
- [ ] Bug fix (non-breaking change which fixes an issue with code or algorithm)
- [x] New feature (non-breaking change which adds functionality to code)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Config and build (change in the configuration and build system, has no impact on code or features)
- [ ] Dependencies (update dependencies and changes associated, has no impact on code or features)
- [ ] Unit Tests (add new Unit Test(s) or improved existing one(s), has no impact on code or features)
- [ ] Documentation (changes or updates in the documentation, has no impact on code or features)

## Checklist:

- [ ] My code follows the code style of this project (only if there are changes in source code).
- [ ] My changes require an update to the documentation (there are changes that require the docs website to be updated).
- [ ] I have updated the documentation accordingly (the changes require an update on the docs in this repo).
- [ ] I have read the [CONTRIBUTING](https://github.com/nanoframework/.github/blob/master/CONTRIBUTING.md) document.
- [ ] I have tested everything locally and all new and existing tests passed (only if there are changes in source code).
- [ ] I have added new tests to cover my changes.

### CODE REVIEW

File: Tests/M5Core2TestApp/Program.cs [hunk 1]
```diff
@@ -1,94 +1,110 @@
...
     if ((e.TouchEventCategory & TouchEventCategory.LiftUp) == TouchEventCategory.LiftUp)
     {
-        Debug.WriteLine(StrLiftUp);
-        Console.Write(StrLiftUp);
+        Debug.WriteLine($"{StrTouchEventCatagory}{StrLiftUp}");
+        Console.WriteLine($"{StrTouchEventCatagory}{StrLiftUp}");
     }
 
-    if ((e.TouchEventCategory & TouchEventCategory.DoubleTouch) == TouchEventCategory.DoubleTouch)
+    if ((e.TouchEventCategory & TouchEventCategory.DoubleTouch) == TouchEventCategory.DoubleTouch) //TODO: never seems to fire!
```
Comment (josesimoes) [inline Tests/M5Core2TestApp/Program.cs:85]: This is a limitation of the touch controller used in the Core2: it doesn't support double tap. So this code can be removed.

File: Tests/M5Core2TestApp/Program.cs [hunk 2]
```diff
@@ -1,94 +1,110 @@
...
+    Console.WriteLine("");
+
+    Console.WriteLine($"{StrGyro}:");
+    Console.WriteLine($"  ({mpu6886.GetGyroscope().X:N2},{mpu6886.GetGyroscope().Y:N2},{mpu6886.GetGyroscope().Z:N2})");
+
+    Console.WriteLine($"{StrAcc}:");
+    Console.WriteLine($"  ({mpu6886.GetAccelerometer().X:N2},{mpu6886.GetAccelerometer().Y:N2},{mpu6886.GetAccelerometer().Z:N2})");
+    
+    Debug.WriteLine($"{StrCpuT}{M5Core2.Power.GetInternalTemperature().DegreesCelsius}�C");
+    Console.WriteLine($"{StrCpuT}{M5Core2.Power.GetInternalTemperature().DegreesCelsius}_C"); //TODO: note that we cannot use the proper symbol!
```
Comment (josesimoes) [inline Tests/M5Core2TestApp/Program.cs:101]: OK for adding this warning note, of course. Please follow the code style an move it to the line before, not in the same code line.

**Comment by josesimoes** [comment]
@networkfusion resources (strings included) go into the assembly where they are declared. I can't think on anything on mscorlib that we can do to improve this or other string resource handling. I don't understand what you're thinking/suggesting. I suggest that we go over this on Discord.

File: Tests/M5Core2TestApp/Program.cs (hunk 2 above)
Comment (Ellerbach) [reply Tests/M5Core2TestApp/Program.cs:101]: Note that the symbol can be added in the font. It's just not been extracted. But right, better to write it!

File: Tests/M5Core2TestApp/Program.cs [hunk 3]
```diff
@@ -1,94 +1,110 @@
...
+const string StrCpuT = "CPU Temperature: ";
+
+M5Core2.Vibrate = false; // Ensure set to false as can otherwise cause a long vibrate on deploy!
 M5Core2.InitializeScreen();
 
 Debug.WriteLine("Hello from M5Core2!");
+Console.WriteLine("Hello from M5Core2!");
+Vibrate(500);
 
-const string Ssid = "SSID";
```
Comment (Ellerbach) [inline Tests/M5Core2TestApp/Program.cs:17]: Why do you remove the wifi connection from the test app?

Comment (networkfusion) [reply Tests/M5Core2TestApp/Program.cs:17]: I didnt feel it brought anything to the test, and none of the other test apps had it. I can re-add it if you feel it is necessary.

File: Tests/M5Core2TestApp/Program.cs (hunk 2 above)
Comment (networkfusion) [reply Tests/M5Core2TestApp/Program.cs]: Improved the comment.

File: Tests/M5Core2TestApp/Program.cs (hunk 1 above)
Comment (networkfusion) [reply Tests/M5Core2TestApp/Program.cs]: Improved the comment.

File: Tests/M5Core2TestApp/Program.cs (hunk 3 above)
Comment (Ellerbach) [reply Tests/M5Core2TestApp/Program.cs:17]: It is needed as one of the pin for touch creates some discrepancies when the wifi is on. It's the only way to test that it's still working properly :-)

Comment (Ellerbach) [reply Tests/M5Core2TestApp/Program.cs:17]: So maybe then add a comment like this and keep the code:

// It is strongly recommended to connect to the wifi while testing the touch interface as wifi is creating discrepancies on the interrupt touch pins.
// This has been managed in the driver but should be tested as well.

Comment (networkfusion) [reply Tests/M5Core2TestApp/Program.cs:17]: Ok. Done.

**Review by Ellerbach — APPROVED** [review]

**Comment by Ellerbach** [comment]
Can we make sure not to create a new version of the nuget when it will be merged as it's just changes on the test application?

**Comment by josesimoes** [comment]
@Ellerbach ✅ 👍🏻

**Pull request merged**